### PR TITLE
[#64] Adds TypeScript support and converts one simple component

### PR DESCRIPTION
**Pull request by justafish**
## Issue

https://github.com/jsdrupal/drupal-admin-ui/issues/64

### CODE REVIEW

File: packages/admin-ui/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    "resolveJsonModule": true,
+    "isolatedModules": true,
+    "noEmit": true,
+    "jsx": "preserve"
+  },
+  "include": [
+    "src"
+  ],
+  "exclude": [
+    "src/tests"
```
Comment (dawehner) [inline packages/admin-ui/tsconfig.json]: Why would we exclude tests?

File: packages/admin-ui/src/NoMatch.tsx [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
+    [pathname: string]: string,
+  },
+}
 
+class NoMatch extends Component<Props> {
   componentDidMount() {
     // We want to redirect everything without a match back to Drupal.
     if (!Object.keys(routes).includes(this.props.location.pathname)) {
-      window.location =
+      window.location.href =
```
Comment (dawehner) [inline packages/admin-ui/src/NoMatch.tsx]: Isn't that doing the same?

File: packages/admin-ui/tsconfig.json (hunk 1 above)
Comment (justafish) [reply packages/admin-ui/tsconfig.json]: The nightwatch tests run on node, and does things like require-ing babel-register, which causes CRA to throw errors

File: packages/admin-ui/src/NoMatch.tsx (hunk 2 above)
Comment (justafish) [reply packages/admin-ui/src/NoMatch.tsx]: Apparently not https://stackoverflow.com/questions/13106950/set-window-location-with-typescript

Comment (dawehner) [reply packages/admin-ui/src/NoMatch.tsx]: Oh I see, this is TS speciality

**Comment by dawehner** [comment]
> I have played with this PR and it looks good... in a moment of paranoia I checked that the build size was unchanged. That is true but as with the typescript monster branch ... the build time increases significantly for me almost doubles.

As long its the production build time, that's totally fine IMHO.

File: packages/admin-ui/src/NoMatch.tsx (hunk 2 above)
Comment (dawehner) [reply packages/admin-ui/src/NoMatch.tsx]: Thank you

File: packages/admin-ui/tsconfig.json (hunk 1 above)
Comment (dawehner) [reply packages/admin-ui/tsconfig.json]: I'm wondering what this mean for unit tests, can they call their appropriate TS alternatives? Anyway, we can always iterate on that

**Comment by dawehner** [comment]
Thank you @martinfrances107 for giving a +1

**Pull request merged**